src/interceptors/XMLHttpRequest/XMLHttpRequestController.ts

```diff
@@ -15,6 +15,7 @@ import { uuidv4 } from '../../utils/uuid'
 import { createResponse } from './utils/createResponse'
 
 const IS_MOCKED_RESPONSE = Symbol('isMockedResponse')
+const isBrowser = typeof window !== 'undefined'
```

Comment:
The presence of `window`, alas, doesn't guarantee this is a browser process. 

Please, could you use [`is-node-process`](https://github.com/mswjs/is-node-process) instead? It's an in-house library we built that reliably distinguishes between actual browsers, browser-like processes (like Electron or JSDOM), and Node.js processes.

```js
import { isNodeProcess } from 'is-node-process'

const IS_NODE = isNodeProcess()

if (IS_NODE) {
  // the logic that sets "X-Request-Id"
}
```

Reply:
No problem. I've changed the implementation via `is-node-process`